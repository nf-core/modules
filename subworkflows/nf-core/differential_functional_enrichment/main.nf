
//
// Perform enrichment analysis
//
include { GPROFILER2_GOST          } from "../../../modules/nf-core/gprofiler2/gost/main.nf"
include { CUSTOM_TABULARTOGSEAGCT  } from '../../../modules/nf-core/custom/tabulartogseagct/main.nf'
include { CUSTOM_TABULARTOGSEACLS  } from '../../../modules/nf-core/custom/tabulartogseacls/main.nf'
include { CUSTOM_TABULARTOGSEACHIP } from '../../../modules/nf-core/custom/tabulartogseachip/main.nf'
include { GSEA_GSEA                } from '../../../modules/nf-core/gsea/gsea/main.nf'
include { PROPR_GREA               } from "../../../modules/nf-core/propr/grea/main.nf"

// Combine meta maps, including merging non-identical values of shared keys (e.g. 'id')
def mergeMaps(meta, meta2){
    (meta + meta2).collectEntries { k, v ->
        meta[k] && meta[k] != v ? [k, "${meta[k]}_${v}"] : [k, v]
    }
}

workflow DIFFERENTIAL_FUNCTIONAL_ENRICHMENT {
    take:
    // input data for functional analysis
    // Note that genesets and background are optional depending on the method.
    // Please set to [] if not provided, eg: [meta, input, [], [], method]
    ch_input                 // [ meta_input, input file, genesets file, background file, method to run ]

    // other - for the moment these files are only needed for GSEA
    // as it is the only one that takes expression data as input
    // if in the future this setting is changed, this section could be removed
    ch_contrasts             // [ meta_contrast, contrast_variable, reference, target ]
    ch_samplesheet           // [ meta_exp, samples sheet ]
    ch_featuresheet          // [ meta_exp, features sheet, features id, features symbol ]

    main:

    ch_versions = Channel.empty()

    // Add method information into meta map of ch_input
    // This information is used later to determine which method to run for each input
    // Also, reorganize the structure to match them with the modules' input organization

    ch_input = ch_input
        .multiMap {
            meta_input, file, genesets, background, analysis_method ->
            def meta_new = meta_input + [ 'method': analysis_method ]
            input:
                [ meta_new, file ]
            genesets:
                [ meta_new, genesets ]  // NOTE here we assume that the modules will not make use of meta_genesets and meta_background
            background:
                [ meta_new, background ]
        }

    // In the case of GSEA, it needs additional files coming from other channels that other methods don't use
    // here we define the input channel for the GSEA section

    def criteria = multiMapCriteria { meta_input, input, genesets, meta_exp, samplesheet, featuresheet, features_id, features_symbol, meta_contrasts, variable, reference, target ->
        def meta_contrasts_new = meta_contrasts + [ 'variable': variable, 'reference': reference, 'target': target ]  // make sure variable, reference, target are in the meta
        def meta_all = mergeMaps(meta_contrasts_new, meta_input)
        input:
            [ meta_all, input ]
        genesets:
            [ meta_all, genesets ]
        contrasts_and_samples:
            [ meta_all, samplesheet ]
        features:
            [ meta_exp, featuresheet ]
        features_cols:
            [ features_id, features_symbol ]
    }
    ch_preinput_for_gsea = ch_input.input
        .join(ch_input.genesets)
        .filter{ it[0].method == 'gsea' }
        .combine(ch_samplesheet.join(ch_featuresheet))
        .combine(ch_contrasts)
        .multiMap(criteria)

    // ----------------------------------------------------
    // Perform enrichment analysis with gprofiler2
    // ----------------------------------------------------

    GPROFILER2_GOST(
        ch_input.input.filter{ it[0].method == 'gprofiler2' },
        ch_input.genesets.filter{ it[0].method == 'gprofiler2'},
        ch_input.background.filter{ it[0].method == 'gprofiler2'}
    )

    // ----------------------------------------------------
    // Perform enrichment analysis with GSEA
    // ----------------------------------------------------

    // NOTE that GCT input can be more than 1, if they come from different tools (eg. limma, deseq2).
    // CLS input can be as many as combinations of input x contrasts
    // Whereas features can be only one file.

    CUSTOM_TABULARTOGSEAGCT(ch_preinput_for_gsea.input)

    CUSTOM_TABULARTOGSEACLS(ch_preinput_for_gsea.contrasts_and_samples)

    CUSTOM_TABULARTOGSEACHIP(
        ch_preinput_for_gsea.features.first(),
        ch_preinput_for_gsea.features_cols.first()
    )

    ch_input_for_gsea = CUSTOM_TABULARTOGSEAGCT.out.gct
        .join(CUSTOM_TABULARTOGSEACLS.out.cls)
        .join( ch_preinput_for_gsea.genesets )

    GSEA_GSEA(
        ch_input_for_gsea,
        ch_input_for_gsea.map{ tuple(it[0].reference, it[0].target) },
        CUSTOM_TABULARTOGSEACHIP.out.chip.first()
    )

    // ----------------------------------------------------
    // Perform enrichment analysis with GREA
    // ----------------------------------------------------

    PROPR_GREA(
        ch_input.input.filter{ it[0].method == 'grea' },
        ch_input.genesets.filter{ it[0].method == 'grea' }
    )

    // collect versions info
    ch_versions = ch_versions
        .mix(GPROFILER2_GOST.out.versions)
        .mix(CUSTOM_TABULARTOGSEAGCT.out.versions)
        .mix(CUSTOM_TABULARTOGSEACLS.out.versions)
        .mix(CUSTOM_TABULARTOGSEACHIP.out.versions)
        .mix(GSEA_GSEA.out.versions)
        .mix(PROPR_GREA.out.versions)

    emit:
    // here we emit the outputs that will be useful afterwards in the
    // nf-core/differentialabundance pipeline

    // gprofiler2-specific outputs
    gprofiler2_all_enrich = GPROFILER2_GOST.out.all_enrich
    gprofiler2_sub_enrich = GPROFILER2_GOST.out.sub_enrich
    gprofiler2_plot_html  = GPROFILER2_GOST.out.plot_html

    // gsea-specific outputs
    gsea_report           = GSEA_GSEA.out.report_tsvs_ref.join(GSEA_GSEA.out.report_tsvs_target)

    // grea-specific outputs
    grea_results          = PROPR_GREA.out.results

    // tool versions
    versions              = ch_versions
}
