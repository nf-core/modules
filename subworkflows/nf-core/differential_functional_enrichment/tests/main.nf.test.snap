{
    "deseq2|limmavoom|propd + gprofiler2|gsea|gsea - mouse": {
        "content": [
            [
                [
                    {
                        "id": "treatment_mCherry_hND6__test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "",
                        "fc_threshold": 1.5,
                        "stat_threshold": 0.05,
                        "method_de": "deseq2",
                        "method": "gprofiler2"
                    },
                    "treatment_mCherry_hND6__test.gprofiler2.all_enriched_pathways.tsv:md5,d586d92fb6cc3e41d493d14c4d9a7d92"
                ],
                [
                    {
                        "id": "treatment_mCherry_hND6__test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "",
                        "fc_threshold": 1.5,
                        "stat_threshold": 0.05,
                        "method_de": "limma",
                        "method": "gprofiler2"
                    },
                    "treatment_mCherry_hND6__test.gprofiler2.all_enriched_pathways.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    {
                        "id": "treatment_mCherry_hND6__test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "",
                        "fc_threshold": 1.5,
                        "stat_threshold": 100,
                        "method_de": "propd",
                        "method": "gprofiler2"
                    },
                    "treatment_mCherry_hND6__test.gprofiler2.all_enriched_pathways.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    {
                        "id": "treatment_mCherry_hND6_sample_number_test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "sample_number",
                        "fc_threshold": 1.5,
                        "stat_threshold": 0.05,
                        "method_de": "deseq2",
                        "method": "gprofiler2"
                    },
                    "treatment_mCherry_hND6_sample_number_test.gprofiler2.all_enriched_pathways.tsv:md5,fa880c70d06f56d8dad0ac8078e6233c"
                ],
                [
                    {
                        "id": "treatment_mCherry_hND6_sample_number_test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "sample_number",
                        "fc_threshold": 1.5,
                        "stat_threshold": 0.05,
                        "method_de": "limma",
                        "method": "gprofiler2"
                    },
                    "treatment_mCherry_hND6_sample_number_test.gprofiler2.all_enriched_pathways.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    {
                        "id": "treatment_mCherry_hND6_sample_number_test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "sample_number",
                        "fc_threshold": 1.5,
                        "stat_threshold": 100,
                        "method_de": "propd",
                        "method": "gprofiler2"
                    },
                    "treatment_mCherry_hND6_sample_number_test.gprofiler2.all_enriched_pathways.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "treatment_mCherry_hND6__test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "",
                        "fc_threshold": 1.5,
                        "stat_threshold": 0.05,
                        "method_de": "deseq2",
                        "method": "gprofiler2"
                    },
                    [
                        "treatment_mCherry_hND6__test.gprofiler2.GO:BP.sub_enriched_pathways.tsv:md5,4c95059438af872253aa95a5ff127fca",
                        "treatment_mCherry_hND6__test.gprofiler2.GO:CC.sub_enriched_pathways.tsv:md5,b4e4009e067f723694efc1a46a609e2b",
                        "treatment_mCherry_hND6__test.gprofiler2.GO:MF.sub_enriched_pathways.tsv:md5,ece72606ce6d1d037cfd5f5b198233a8",
                        "treatment_mCherry_hND6__test.gprofiler2.REAC.sub_enriched_pathways.tsv:md5,dc7b6f67903e25076ab95a7e1bb39bbe"
                    ]
                ],
                [
                    {
                        "id": "treatment_mCherry_hND6_sample_number_test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "sample_number",
                        "fc_threshold": 1.5,
                        "stat_threshold": 0.05,
                        "method_de": "deseq2",
                        "method": "gprofiler2"
                    },
                    [
                        "treatment_mCherry_hND6_sample_number_test.gprofiler2.GO:BP.sub_enriched_pathways.tsv:md5,dc27af1c941636f8b03e8c8724773725",
                        "treatment_mCherry_hND6_sample_number_test.gprofiler2.GO:CC.sub_enriched_pathways.tsv:md5,09b7770d59d2efa8a58f300a16a41ae7",
                        "treatment_mCherry_hND6_sample_number_test.gprofiler2.GO:MF.sub_enriched_pathways.tsv:md5,a261d658ca1b1ec0511e55da734d7810",
                        "treatment_mCherry_hND6_sample_number_test.gprofiler2.KEGG.sub_enriched_pathways.tsv:md5,3cbdbbb1c5b232ab73d35d4a0953d19c",
                        "treatment_mCherry_hND6_sample_number_test.gprofiler2.REAC.sub_enriched_pathways.tsv:md5,8472cb5792bf7ec044bb95df356ad47e"
                    ]
                ]
            ],
            [
                "treatment_mCherry_hND6__test.gprofiler2.gostplot.html",
                "treatment_mCherry_hND6_sample_number_test.gprofiler2.gostplot.html"
            ],
            [
                [
                    {
                        "id": "treatment_mCherry_hND6__test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "",
                        "method_de": "deseq2",
                        "method": "gsea"
                    },
                    "treatment_mCherry_hND6__test.mh.all.v2022.1.Mm.symbols.gsea_report_for_mCherry.tsv:md5,ce8792382ae299749445767ff16aaecc",
                    "treatment_mCherry_hND6__test.mh.all.v2022.1.Mm.symbols.gsea_report_for_hND6.tsv:md5,354a961c7e1417db2bf1f7e8d00c54f0"
                ],
                [
                    {
                        "id": "treatment_mCherry_hND6__test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "",
                        "method_de": "limma",
                        "method": "gsea"
                    },
                    "treatment_mCherry_hND6__test.mh.all.v2022.1.Mm.symbols.gsea_report_for_mCherry.tsv:md5,ce8792382ae299749445767ff16aaecc",
                    "treatment_mCherry_hND6__test.mh.all.v2022.1.Mm.symbols.gsea_report_for_hND6.tsv:md5,a4dcc9581c9c63d35ef32ee0df882074"
                ],
                [
                    {
                        "id": "treatment_mCherry_hND6_sample_number_test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "sample_number",
                        "method_de": "deseq2",
                        "method": "gsea"
                    },
                    "treatment_mCherry_hND6_sample_number_test.mh.all.v2022.1.Mm.symbols.gsea_report_for_mCherry.tsv:md5,ce8792382ae299749445767ff16aaecc",
                    "treatment_mCherry_hND6_sample_number_test.mh.all.v2022.1.Mm.symbols.gsea_report_for_hND6.tsv:md5,354a961c7e1417db2bf1f7e8d00c54f0"
                ],
                [
                    {
                        "id": "treatment_mCherry_hND6_sample_number_test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "sample_number",
                        "method_de": "limma",
                        "method": "gsea"
                    },
                    "treatment_mCherry_hND6_sample_number_test.mh.all.v2022.1.Mm.symbols.gsea_report_for_mCherry.tsv:md5,ce8792382ae299749445767ff16aaecc",
                    "treatment_mCherry_hND6_sample_number_test.mh.all.v2022.1.Mm.symbols.gsea_report_for_hND6.tsv:md5,a4dcc9581c9c63d35ef32ee0df882074"
                ]
            ],
            [
                [
                    {
                        "id": "treatment_mCherry_hND6__test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "",
                        "method_de": "propd",
                        "method": "grea"
                    },
                    "treatment_mCherry_hND6__test.grea.tsv:md5,786faeccf39926d2f7c980ef549a2697"
                ],
                [
                    {
                        "id": "treatment_mCherry_hND6_sample_number_test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "sample_number",
                        "method_de": "propd",
                        "method": "grea"
                    },
                    "treatment_mCherry_hND6_sample_number_test.grea.tsv:md5,786faeccf39926d2f7c980ef549a2697"
                ]
            ],
            [
                "versions.yml:md5,25ab98049a601f4940f3e5a24aa73f55",
                "versions.yml:md5,25ab98049a601f4940f3e5a24aa73f55",
                "versions.yml:md5,25ab98049a601f4940f3e5a24aa73f55",
                "versions.yml:md5,25ab98049a601f4940f3e5a24aa73f55",
                "versions.yml:md5,7861e3047b941b86ef50124168a13b51",
                "versions.yml:md5,7861e3047b941b86ef50124168a13b51",
                "versions.yml:md5,7861e3047b941b86ef50124168a13b51",
                "versions.yml:md5,7861e3047b941b86ef50124168a13b51",
                "versions.yml:md5,7861e3047b941b86ef50124168a13b51",
                "versions.yml:md5,7861e3047b941b86ef50124168a13b51",
                "versions.yml:md5,cd9cd1563a983e586b15fd2276da8bfb",
                "versions.yml:md5,cd9cd1563a983e586b15fd2276da8bfb",
                "versions.yml:md5,f2db818ec8143f64399247548098b643",
                "versions.yml:md5,f3be00003ea786284231f77c03c92eed",
                "versions.yml:md5,f3be00003ea786284231f77c03c92eed",
                "versions.yml:md5,f3be00003ea786284231f77c03c92eed",
                "versions.yml:md5,f3be00003ea786284231f77c03c92eed",
                "versions.yml:md5,f823a4e0a4d8744b89aff1391ca74e3a",
                "versions.yml:md5,f823a4e0a4d8744b89aff1391ca74e3a",
                "versions.yml:md5,f823a4e0a4d8744b89aff1391ca74e3a",
                "versions.yml:md5,f823a4e0a4d8744b89aff1391ca74e3a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-24T15:12:40.203142527"
    },
    "propd + grea - mouse": {
        "content": [
            [
                [
                    {
                        "id": "treatment_mCherry_hND6__test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "",
                        "method_de": "propd",
                        "method": "grea"
                    },
                    "treatment_mCherry_hND6__test.grea.tsv:md5,786faeccf39926d2f7c980ef549a2697"
                ],
                [
                    {
                        "id": "treatment_mCherry_hND6_sample_number_test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "sample_number",
                        "method_de": "propd",
                        "method": "grea"
                    },
                    "treatment_mCherry_hND6_sample_number_test.grea.tsv:md5,786faeccf39926d2f7c980ef549a2697"
                ]
            ],
            {
                "DIFFERENTIAL_FUNCTIONAL_ENRICHMENT:PROPR_GREA": {
                    "r-propr": "5.1.5"
                }
            },
            [
                "versions.yml:md5,cd9cd1563a983e586b15fd2276da8bfb",
                "versions.yml:md5,cd9cd1563a983e586b15fd2276da8bfb"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-24T15:10:22.944909501"
    },
    "deseq2 + gprofiler2 - mouse": {
        "content": [
            [
                [
                    {
                        "id": "treatment_mCherry_hND6__test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "",
                        "fc_threshold": 1.5,
                        "stat_threshold": 0.05,
                        "method_de": "deseq2",
                        "method": "gprofiler2"
                    },
                    "treatment_mCherry_hND6__test.gprofiler2.all_enriched_pathways.tsv:md5,d586d92fb6cc3e41d493d14c4d9a7d92"
                ],
                [
                    {
                        "id": "treatment_mCherry_hND6_sample_number_test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "sample_number",
                        "fc_threshold": 1.5,
                        "stat_threshold": 0.05,
                        "method_de": "deseq2",
                        "method": "gprofiler2"
                    },
                    "treatment_mCherry_hND6_sample_number_test.gprofiler2.all_enriched_pathways.tsv:md5,fa880c70d06f56d8dad0ac8078e6233c"
                ]
            ],
            [
                [
                    {
                        "id": "treatment_mCherry_hND6__test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "",
                        "fc_threshold": 1.5,
                        "stat_threshold": 0.05,
                        "method_de": "deseq2",
                        "method": "gprofiler2"
                    },
                    [
                        "treatment_mCherry_hND6__test.gprofiler2.GO:BP.sub_enriched_pathways.tsv:md5,4c95059438af872253aa95a5ff127fca",
                        "treatment_mCherry_hND6__test.gprofiler2.GO:CC.sub_enriched_pathways.tsv:md5,b4e4009e067f723694efc1a46a609e2b",
                        "treatment_mCherry_hND6__test.gprofiler2.GO:MF.sub_enriched_pathways.tsv:md5,ece72606ce6d1d037cfd5f5b198233a8",
                        "treatment_mCherry_hND6__test.gprofiler2.REAC.sub_enriched_pathways.tsv:md5,dc7b6f67903e25076ab95a7e1bb39bbe"
                    ]
                ],
                [
                    {
                        "id": "treatment_mCherry_hND6_sample_number_test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "sample_number",
                        "fc_threshold": 1.5,
                        "stat_threshold": 0.05,
                        "method_de": "deseq2",
                        "method": "gprofiler2"
                    },
                    [
                        "treatment_mCherry_hND6_sample_number_test.gprofiler2.GO:BP.sub_enriched_pathways.tsv:md5,dc27af1c941636f8b03e8c8724773725",
                        "treatment_mCherry_hND6_sample_number_test.gprofiler2.GO:CC.sub_enriched_pathways.tsv:md5,09b7770d59d2efa8a58f300a16a41ae7",
                        "treatment_mCherry_hND6_sample_number_test.gprofiler2.GO:MF.sub_enriched_pathways.tsv:md5,a261d658ca1b1ec0511e55da734d7810",
                        "treatment_mCherry_hND6_sample_number_test.gprofiler2.KEGG.sub_enriched_pathways.tsv:md5,3cbdbbb1c5b232ab73d35d4a0953d19c",
                        "treatment_mCherry_hND6_sample_number_test.gprofiler2.REAC.sub_enriched_pathways.tsv:md5,8472cb5792bf7ec044bb95df356ad47e"
                    ]
                ]
            ],
            [
                "treatment_mCherry_hND6__test.gprofiler2.gostplot.html",
                "treatment_mCherry_hND6_sample_number_test.gprofiler2.gostplot.html"
            ],
            {
                "DIFFERENTIAL_FUNCTIONAL_ENRICHMENT:GPROFILER2_GOST": {
                    "r-base": "4.3.3",
                    "r-ggplot2": "3.4.3",
                    "r-gprofiler2": "0.2.2"
                }
            },
            [
                "versions.yml:md5,7861e3047b941b86ef50124168a13b51",
                "versions.yml:md5,7861e3047b941b86ef50124168a13b51"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-24T15:08:35.939943561"
    },
    "test gprofiler2 - mouse": {
        "content": [
            [
                [
                    {
                        "id": "Condition_genotype_WT_KO",
                        "variable": "Condition genotype",
                        "reference": "WT",
                        "target": "KO",
                        "blocking": "batch",
                        "method": "gprofiler2"
                    },
                    "Condition_genotype_WT_KO.gprofiler2.all_enriched_pathways.tsv:md5,1134a02ca061c463bcbff277eefbfb19"
                ]
            ],
            [
                [
                    {
                        "id": "Condition_genotype_WT_KO",
                        "variable": "Condition genotype",
                        "reference": "WT",
                        "target": "KO",
                        "blocking": "batch",
                        "method": "gprofiler2"
                    },
                    [
                        "Condition_genotype_WT_KO.gprofiler2.GO:BP.sub_enriched_pathways.tsv:md5,d527b94cdb160070bcaa0bfb0cecf914",
                        "Condition_genotype_WT_KO.gprofiler2.GO:CC.sub_enriched_pathways.tsv:md5,ef418c3f06d50446317928e37ec8ddfb",
                        "Condition_genotype_WT_KO.gprofiler2.GO:MF.sub_enriched_pathways.tsv:md5,e460d4917feb0b64d334a528f59e0731",
                        "Condition_genotype_WT_KO.gprofiler2.HP.sub_enriched_pathways.tsv:md5,865d8f092503552831c51d775a98c6eb",
                        "Condition_genotype_WT_KO.gprofiler2.KEGG.sub_enriched_pathways.tsv:md5,413724002abe683f376ea914d4f21ade",
                        "Condition_genotype_WT_KO.gprofiler2.MIRNA.sub_enriched_pathways.tsv:md5,fce81051d7af955ddb2925ba2da9ff57",
                        "Condition_genotype_WT_KO.gprofiler2.REAC.sub_enriched_pathways.tsv:md5,e89e1876698ea644671a0720c85f4dbb",
                        "Condition_genotype_WT_KO.gprofiler2.TF.sub_enriched_pathways.tsv:md5,e272217ec7fcf01ea463ab8bcc8335cf",
                        "Condition_genotype_WT_KO.gprofiler2.WP.sub_enriched_pathways.tsv:md5,fdd1efa836d85bb127e933e925290cba"
                    ]
                ]
            ],
            [
                "Condition_genotype_WT_KO.gprofiler2.gostplot.html"
            ],
            {
                "DIFFERENTIAL_FUNCTIONAL_ENRICHMENT:GPROFILER2_GOST": {
                    "r-base": "4.3.3",
                    "r-ggplot2": "3.4.3",
                    "r-gprofiler2": "0.2.2"
                }
            },
            [
                "versions.yml:md5,7861e3047b941b86ef50124168a13b51"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-24T15:07:42.734133807"
    },
    "deseq2 + gsea - mouse": {
        "content": [
            [
                [
                    {
                        "id": "treatment_mCherry_hND6__test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "",
                        "method_de": "deseq2",
                        "method": "gsea"
                    },
                    "treatment_mCherry_hND6__test.mh.all.v2022.1.Mm.symbols.gsea_report_for_mCherry.tsv:md5,ce8792382ae299749445767ff16aaecc",
                    "treatment_mCherry_hND6__test.mh.all.v2022.1.Mm.symbols.gsea_report_for_hND6.tsv:md5,354a961c7e1417db2bf1f7e8d00c54f0"
                ],
                [
                    {
                        "id": "treatment_mCherry_hND6_sample_number_test",
                        "variable": "treatment",
                        "reference": "mCherry",
                        "target": "hND6",
                        "blocking": "sample_number",
                        "method_de": "deseq2",
                        "method": "gsea"
                    },
                    "treatment_mCherry_hND6_sample_number_test.mh.all.v2022.1.Mm.symbols.gsea_report_for_mCherry.tsv:md5,ce8792382ae299749445767ff16aaecc",
                    "treatment_mCherry_hND6_sample_number_test.mh.all.v2022.1.Mm.symbols.gsea_report_for_hND6.tsv:md5,354a961c7e1417db2bf1f7e8d00c54f0"
                ]
            ],
            {
                "DIFFERENTIAL_FUNCTIONAL_ENRICHMENT:GSEA_GSEA": {
                    "gsea": "4.3.2"
                }
            },
            [
                "versions.yml:md5,25ab98049a601f4940f3e5a24aa73f55",
                "versions.yml:md5,25ab98049a601f4940f3e5a24aa73f55",
                "versions.yml:md5,f2db818ec8143f64399247548098b643",
                "versions.yml:md5,f3be00003ea786284231f77c03c92eed",
                "versions.yml:md5,f3be00003ea786284231f77c03c92eed",
                "versions.yml:md5,f823a4e0a4d8744b89aff1391ca74e3a",
                "versions.yml:md5,f823a4e0a4d8744b89aff1391ca74e3a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-24T15:09:55.875311067"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Condition_genotype_WT_KO",
                            "variable": "Condition genotype",
                            "reference": "WT",
                            "target": "KO",
                            "blocking": "batch",
                            "method": "gprofiler2"
                        },
                        "Condition_genotype_WT_KO.gprofiler2.all_enriched_pathways.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "Condition_genotype_WT_KO",
                            "variable": "Condition genotype",
                            "reference": "WT",
                            "target": "KO",
                            "blocking": "batch",
                            "method": "gprofiler2"
                        },
                        "Condition_genotype_WT_KO.gprofiler2.*.sub_enriched_pathways.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "Condition_genotype_WT_KO",
                            "variable": "Condition genotype",
                            "reference": "WT",
                            "target": "KO",
                            "blocking": "batch",
                            "method": "gprofiler2"
                        },
                        "Condition_genotype_WT_KO.gprofiler2.gostplot.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    "versions.yml:md5,7861e3047b941b86ef50124168a13b51"
                ],
                "gprofiler2_all_enrich": [
                    [
                        {
                            "id": "Condition_genotype_WT_KO",
                            "variable": "Condition genotype",
                            "reference": "WT",
                            "target": "KO",
                            "blocking": "batch",
                            "method": "gprofiler2"
                        },
                        "Condition_genotype_WT_KO.gprofiler2.all_enriched_pathways.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "gprofiler2_plot_html": [
                    [
                        {
                            "id": "Condition_genotype_WT_KO",
                            "variable": "Condition genotype",
                            "reference": "WT",
                            "target": "KO",
                            "blocking": "batch",
                            "method": "gprofiler2"
                        },
                        "Condition_genotype_WT_KO.gprofiler2.gostplot.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "gprofiler2_sub_enrich": [
                    [
                        {
                            "id": "Condition_genotype_WT_KO",
                            "variable": "Condition genotype",
                            "reference": "WT",
                            "target": "KO",
                            "blocking": "batch",
                            "method": "gprofiler2"
                        },
                        "Condition_genotype_WT_KO.gprofiler2.*.sub_enriched_pathways.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "grea_results": [
                    
                ],
                "gsea_report": [
                    
                ],
                "versions": [
                    "versions.yml:md5,7861e3047b941b86ef50124168a13b51"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-24T17:38:14.74535"
    }
}